# Basics

In [1]:
print("hi good morning welcome to data science")

hi good morning welcome to data science


In [1]:
print("hi welcome to europe")

hi welcome to europe


In [2]:
print("jjjjj)

SyntaxError: EOL while scanning string literal (<ipython-input-2-44de8066e0ee>, line 1)

# Double and single Quotes

In [4]:
print("Romantic language of is Italian")

Romantic language of is Italian


In [5]:
print('Romantic Country is Italian for lovers')

Romantic Country is Italian for lovers


# python 2.x version print statement

In [6]:
print "University of Naples"

SyntaxError: Missing parentheses in call to 'print'. Did you mean print("University of Naples")? (<ipython-input-6-368ea315617e>, line 1)

# how to comment in python by using hash symbol

In [7]:
#hii this print statement

# comment for multiple line of code bys using ------> triple quotes(single & double quotes) 
both will works the same no difference in that

In [9]:
"""
hi
dd
f
g
jj
kk
ll
"""

'\nhi\ndd\nf\ng\njj\nkk\nll\n'

In [10]:
'''
gg
hh
jj
kk'''

'\ngg\nhh\njj\nkk'

why multiple line comment works as output because we are not giving anything below to complie or debuging the code so default by it prints the multiple line comment unless and until we are providing the code to debug it

here in below code i provided the code for the above explain with proff

In [146]:
"""
Here am going to print
multiple line comments 
for sum of two numbers
"""

a = 10
b = 20
c = a+b
print(c)

30


# variable 

In [12]:
a=10  #storing not execute

In [15]:
a #print

10

In [16]:
type(a)

int

In [17]:
a="ddddd"
b="hhhhh"

In [18]:
a,b="dddd","hhhh"

In [19]:
print(a)

dddd


In [20]:
print(b)

hhhh


In [21]:
print(a,b)

dddd hhhh


In [22]:
type(a)

str

In [23]:
type(b)

str

In [24]:
c=10000

In [25]:
print(c)

10000


In [26]:
type(c)

int

In [27]:
print(a,b,c)

dddd hhhh 10000


In [29]:
print(a,b,c,"all 3 variable declaration")

dddd hhhh 10000 all 3 variable declaration


variable decalation in python

small letter

capital letter

underscore in starting

In [30]:
b=10

In [31]:
b

10

In [32]:
B=20

In [33]:
B

20

In [34]:
_a=20

In [35]:
_a

20

In [40]:
f^=10
f^

SyntaxError: invalid syntax (<ipython-input-40-d04a9fe24220>, line 2)

In [42]:
k)=200

SyntaxError: unmatched ')' (<ipython-input-42-0a3f1d5ab8da>, line 1)

In [44]:
a&="10"

TypeError: unsupported operand type(s) for &=: 'str' and 'str'

# Best Practice variable

when we are creatig a vaiable someone or others shold able to understand easily

In [45]:
user = "Abhishek"
userID = "4AI14CS002"
email = "wwww@gmail.com"
password = "123456789"

In [49]:
print(user,userID,email,password)

NameError: name 'userID' is not defined

Boolean

In [50]:
a=True
b=False

In [51]:
print(a)

True


In [52]:
type(a)

bool

In [53]:
type(b)

bool

In [54]:
print(True + True + True + False +True)

4


In [55]:
print(False - True + True)

0


In [57]:
print(True+True)

2


In [59]:
print(False+False)

0


True  = 1

False = 0

this is by default of python programming 

In [60]:
True / False

ZeroDivisionError: division by zero

in core python programming language True/False it gives error i.e, ZeroDivisionError: division by zero

    But in numpy library it gives "infin" 

In [61]:
1/0

ZeroDivisionError: division by zero

In [62]:
a = 2+3j

In [63]:
a

(2+3j)

here 2 is real number

3 is imaginary number and j is notation

In [64]:
type(a)

complex

In [65]:
a.real

2.0

In [66]:
a.imag

3.0

In [68]:
1 + "abhi" 

TypeError: unsupported operand type(s) for +: 'int' and 'str'

Python is dynamic programming language

In [69]:
1+1.3

2.3

In [70]:
1+"abhis"

TypeError: unsupported operand type(s) for +: 'int' and 'str'

because of different types of data for this we want to use type casting i.e, strong casting

In [71]:
'1' + "abhis"

'1abhis'

In [72]:
100 + 200.67

300.66999999999996

this above is upcasting by dynamic programming from int to float number

In [73]:
abhi * 3

NameError: name 'abhi' is not defined

here there is again error because of type error

In [74]:
"abhi" * 3

'abhiabhiabhi'

In [75]:
int('20')

20

In [76]:
int("abhi")

ValueError: invalid literal for int() with base 10: 'abhi'

In [81]:
a=10

# input

In [82]:
a=input()

barandooru


In [83]:
a

'barandooru'

In [84]:
a = input("enter a number")

enter a number12345


In [85]:
a

'12345'

In [86]:
type(a)

str

# tricky simple problem 
Take 3 input of integer 

perform (a+b)/c and store it one variable

add complex number for that variable

add  name number for that variable

In [97]:
a=int(input("enter a number "))
b=int(input("enter b number "))
c=int(input("enter a number "))

enter a number 1
enter b number 2
enter a number 3


In [113]:
a,b,c

(1, 2, 3)

In [114]:
o = (a+b)/c

In [115]:
print(o)

1.0


In [116]:
i = o+2+3j

In [117]:
print(i)

(3+3j)


In [120]:
str(i) + "barandooru"

'(3+3j)barandooru'

# if 
when we compare with java, c and dot net so on....
there we used curley braces for the for excute the loop but in python we does't required to execute the curely braces because the logic behind is 

how it execute the logic in if statement? means 

Based on indentation of logic code
--------------------------------------------------
in below code you can observe it in print statement of indent in if loop

In [124]:
a = int(input("enter a number less than 25 to validate: "))
if a < 25:

SyntaxError: unexpected EOF while parsing (<ipython-input-124-7245e9eb1b53>, line 2)

# pass

In [125]:
a = int(input("enter a number less than 25 to validate: "))
if a < 25:
    pass

enter a number less than 25 to validate: 55


In [122]:
a = int(input("enter a number less than 25 to validate: "))
if a < 25:
    print("number is less than 25 ")

enter a number less than 25 to validate: 22
number is less than 25 


In [123]:
a = int(input("enter a number less than 25 to validate: "))
if a < 25:
print("number is less than 25 ")

IndentationError: expected an indented block (<ipython-input-123-33a2af986a23>, line 3)

why the above code is error means -----------> print is considered as sepertate statement sooo because it follow indentation

# else

In [126]:
num = int(input("enter a number "))
if(num>25):
    print("number is greater than 25")
else:
    print("number is lesser than 25")

enter a number 22
number is lesser than 25


# elif

In [133]:
num = int(input("enter a number: "))
if num<25:
    print("number is less than 25")
elif num<=100:
    print("number is less than or equals to 100")
else:
    print("number is greater than 100: ",num)

enter a number: 100
number is less than or equals to 100


What is the output of below code? think and give logical reason and why?

In [140]:
salary = int(input("enter salary: "))
if salary<250000:
    print("senior enginner")
elif salary<50000:
    print("junior enginner")
elif(salary<10000):
    print("Associate engineer")
else:
    print("Board Memeber")

enter salary: 230
senior enginner


here in the above code we can see that it senior enginner salary is 230  why? and it execute line by line in python  

In [144]:
salary = int(input("enter salary: "))
if salary<25000:
    print("junior enginner")
elif salary<50000:
    print("senior enginner")
    if(salary<10000):
        print("managing QA team")
    elif(salary<20000):
        print("Managing Testing team")
    elif(salary<300000):
        print("Managing Developer team")
    else:
        print("Managing all 3 teams")
elif(salary<100000):
    print("consultant")
else:
    print("Board Member")

enter salary: 35000
senior enginner
Managing Developer team


# Identity of an object

In [147]:
a=10
b=20
print(id(a))
print(id(b))

140703541831744
140703541832064


In [148]:
a=30
b=30
print(id(a))
print(id(b))

140703541832384
140703541832384


refer above the difference of identity of object, memory location why difference means a and b are just a varibale for temporary but memoey location of 10 and 20 are stored in memory it is fixed 